README.md

```diff
@@ -17,21 +17,66 @@ pipx install tap-googleads
 
 ## Configuration
 
-### Get refresh token
-1. GET https://accounts.google.com/o/oauth2/v2/auth?response_type=code&client_id=client_id&redirect_uri=http://127.0.0.1&scope=https://www.googleapis.com/auth/adwords&state=autoidm&access_type=offline&prompt=select_account&include_granted_scopes=true
-1. POST https://www.googleapis.com/oauth2/v4/token?code={code}&client_id={client_id}&client_secret={client_secret}&redirect_uri=http://127.0.0.1&grant_type=authorization_code
-1. POST https://www.googleapis.com/oauth2/v4/token?refresh_token={refres_token}&client_id={client_id}&client_secret={client_secret]&grant_type=refresh_token
 ### Accepted Config Options
 
 - [ ] `Developer TODO:` Provide a list of config options accepted by the tap.
 
-A full list of supported settings and capabilities for this
-tap is available by running:
+### This tap supports two sets of configs:
+
+### Using Your Own Credentials
+
+Settings required to run this tap.
+
+- `client_id` (required)
+- `client_secret` (required)
+- `developer_token` (required)
+- `refresh_token` (required)
+- `customer_login_id` (required)
+- `customer_id` (required)
+- `start_date` (optional)
 
+How to get these settings can be found in the following Google Ads documentation:
+
+https://developers.google.com/adwords/api/docs/guides/authentication
+
+If you have installed the tap you can run the following commands to see more information about the tap.
 ```bash
 tap-googleads --about
 ```
 
+### Getting A Refresh Token
+1. GET https://accounts.google.com/o/oauth2/v2/auth?response_type=code&client_id=client_id&redirect_uri=http://127.0.0.1&scope=https://www.googleapis.com/auth/adwords&state=autoidm&access_type=offline&prompt=select_account&include_granted_scopes=true
+1. POST https://www.googleapis.com/oauth2/v4/token?code={code}&client_id={client_id}&client_secret={client_secret}&redirect_uri=http://127.0.0.1&grant_type=authorization_code
+1. POST https://www.googleapis.com/oauth2/v4/token?refresh_token={refresh_token}&client_id={client_id}&client_secret={client_secret}&grant_type=refresh_token
+
+
+### Proxy OAuth Credentials
+
+To run the tap yourself It is highly recommended to use the [Using Your Own Credentials](#using-your-own-credentials) section listed above.
+
+These settings for handling your credentials through a Proxy OAuth Server, these settings are used by default in a [Matatika](https://www.matatika.com/) workspace.
```

Comment:
I believe the end goal from Matatika's end is to hide the `client_secret` / `client_id` from the user of this tap. Is that right?

Reply:
Yeah the use case for us is: 

- A user comes into our app with just a google account and an associated google ads account. (No `client_secret` and `client_id` of their own).
- They give our app access to their google account, specifically access to google ads data.
- Then we can use our app's `client_id` and `client_secret` to get refresh_tokens for them.

We can't have our `client_id` and `client_secret` in the environment as it is the user environment, they can set/update/print out whatever is in there.

So we use a proxy oauth server, verify the users identity with our internal `bearer_token` (which is the users so can be in the environment), pass the `customer_id`, `developer_token` or any other setting the user has provided to the proxy oauth server, which then makes the requests for `refresh_token`(s) ect on behalf of the user.